docs/en/reference/configuration.rst

```diff
@@ -341,8 +341,10 @@ the corresponding platform class.
 You can also pass the ``serverVersion`` option if you want to disable automatic
 database platform detection and choose the platform version implementation explicitly.
 
-If you are running a MariaDB database, you should prefix the ``serverVersion``
-with ``mariadb-`` (ex: ``mariadb-10.2.12``).
+Please specify the full server version as the database server would report it,
+e.g. ``8.0.32`` instead of just ``8`` if your server is a MySQL 8.0.32. This
+means that if you are running a MariaDB database, you should suffix the
+``serverVersion`` with ``-MariaDB`` (ex: ``10.11.2-MariaDB``).
```

Comment:
I'd remove this "you should suffix" part as it's controversial. The user should specify the server version verbatim, without adding or removing anything. The documentation could suggest running the following code snippet and using its output as the value of `serverVersion`:
```php
echo $connection->getServerVersion(), PHP_EOL;
```
Specific versions of MariaDB could be used as an example.

Note that this method is `private` in DBAL 3, so the snippet needs to be adapted.

Reply:
> Since we're talking about MySQL/MariaDB specifically [...]

Are we? This documentation applies to all platforms for which the DBAL has different implementations for different versions.

Reply:
Which is an implementation detail of our version detection logic. Why should users be aware of that?

The whole extent of this feature is providing the server version in the configuration in order to prevent the application from making unnecessary calls to `Connection::getServerVersion()`. The more implementation details we expose via the documentation, the more chances are that the documentation will become obsolete at some point or will have to be continuously changed alongside the changes in supported platforms.

Reply:
A private `getServerVersion()` is also an implementation details in form of a choice to outsource some logic into a private method. Private methods can be removed and shouldn't appear in the docs. 

I'm not sure if we currently have a generic way to get the server version string for the config,

Reply:
It's a good point. It is private because it's only used for platform detection and is not supposed to be used outside of DBAL. But the thing is that we want to call it in order to use by the DBAL internally.

Anyways, I think this issue doesn't deserve that much discussion. I'm fine with whatever you all decide.